#! /usr/bin/env python

PACKAGE='kinematic_retargeting'
import roslib; roslib.load_manifest(PACKAGE)

from math import pi

from driver_base.msg import SensorLevels
from dynamic_reconfigure.parameter_generator import *

gen = ParameterGenerator()

#Name            Type   Reconfiguration level             Description         Default Min Max
gen.add( "ee_weight", double_t, SensorLevels.RECONFIGURE_RUNNING, "Weight placed on end effector in cost computation", 5.0, 0.0, 10.0 )
gen.add( "max_iterations", int_t, SensorLevels.RECONFIGURE_RUNNING, "Retargeting iterations", 500, 1, 10000 )
gen.add( "step_threshold", double_t, SensorLevels.RECONFIGURE_RUNNING, "Retargeting is complete when cost function step size drops below this threshold", 0.0001, 0.0000001, 0.1 )
#gen.add( "normalize_target_chain", bool_t, SensorLevels.RECONFIGURE_RUNNING, "Normalize target chain to have same length as source before retargeting", True)
#gen.add( "show_joint_names", bool_t, SensorLevels.RECONFIGURE_RUNNING, "Publish joint name markers when visualizing chains", False)

exit(gen.generate(PACKAGE, "dynamic_reconfigure_node", "HumanoidRetargeter"))
